diff --git a/src/lib/kuiper/Kuiper.Approximates.Base.fst b/src/lib/kuiper/Kuiper.Approximates.Base.fst
--- a/src/lib/kuiper/Kuiper.Approximates.Base.fst
+++ b/src/lib/kuiper/Kuiper.Approximates.Base.fst
@@ -102,0 +103,4 @@ class floating_real_like (a:Type) {| scalar a, floating a, real_like a |} = {
+
+  sqrt_approx : x:a -> r:FStar.Math.Sqrt.rnonneg ->
+                Lemma (requires v_approximates x r)
+                      (ensures v_approximates (sqrt x) (FStar.Math.Sqrt.sqrt r));
@@ -156,0 +161,10 @@ let log_approx_pat
+
+let sqrt_approx_pat
+  (a:Type) {| scalar a, floating a, real_like a, rr : floating_real_like a |}
+  (x : a) (r : FStar.Math.Sqrt.rnonneg) :
+    Lemma (requires v_approximates x r)
+          (ensures v_approximates (sqrt x) (FStar.Math.Sqrt.sqrt r))
+          [SMTPat (sqrt x);
+           SMTPat (v_approximates x r);
+           SMTPat (has_type rr (floating_real_like a))]
+  = sqrt_approx x r
